common/client/multi_node.go

```diff
@@ -0,0 +1,668 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/smartcontractkit/chainlink/v2/common/chains/client"
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+	"github.com/smartcontractkit/chainlink/v2/core/config"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+var (
+	// PromMultiNodeRPCNodeStates reports current RPC node state
+	PromMultiNodeRPCNodeStates = promauto.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "multi_node_states",
+		Help: "The number of RPC nodes currently in the given state for the given chain",
+	}, []string{"network", "chainId", "state"})
+	ErroringNodeError = fmt.Errorf("no live nodes available")
+)
+
+const (
+	NodeSelectionMode_HighestHead     = "HighestHead"
+	NodeSelectionMode_RoundRobin      = "RoundRobin"
+	NodeSelectionMode_TotalDifficulty = "TotalDifficulty"
+	NodeSelectionMode_PriorityLevel   = "PriorityLevel"
```

Comment:
I don't know what the best way to name these is, but let's drop the un-idiomatic underscore?
```suggestion
	NodeSelectionModeHighestHead     = "HighestHead"
	NodeSelectionModeRoundRobin      = "RoundRobin"
	NodeSelectionModeTotalDifficulty = "TotalDifficulty"
	NodeSelectionModePriorityLevel   = "PriorityLevel"
```
```suggestion
	HighestHeadNodeSelectionMode     = "HighestHead"
	RoundRobinNodeSelectionMode      = "RoundRobin"
	TotalDifficultyNodeSelectionMode = "TotalDifficulty"
	PriorityLevelNodeSelectionMode   = "PriorityLevel"
```
```suggestion
	HighestHeadSelectionMode     = "HighestHead"
	RoundRobinSelectionMode      = "RoundRobin"
	TotalDifficultySelectionMode = "TotalDifficulty"
	PriorityLevelSelectionMode   = "PriorityLevel"
```